test/jdk/java/awt/Robot/ScreenCaptureRobotTest.java

```diff
@@ -0,0 +1,168 @@
+        g.setColor(Color.RED);
+        g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 20));
+        g.drawString("Capture This", 10, 40);
+        g.dispose();
+
+        canvas = new ImageCanvas();
+        canvas.setBackground(Color.YELLOW);
+        frame.setLayout(new BorderLayout());
+        frame.add(canvas);
+        frame.setSize(300, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void doTest() throws Exception {
+        Robot robot = new Robot();
+        robot.waitForIdle();
+        robot.delay(500);
+
+        EventQueue.invokeAndWait(() -> {
+            point = canvas.getLocationOnScreen();
+        });
+
+        Rectangle rect = new Rectangle(point.x + 10, point.y + 10, IMAGE_WIDTH,
+                IMAGE_HEIGHT);
+
+        BufferedImage capturedImage = robot.createScreenCapture(rect);
+
+        if (!compareImages(capturedImage, realImage)) {
+            String errorMessage = "FAIL : Captured Image is different from "
+                    + "the real image";
+            System.err.println("Test failed");
+            saveImage(capturedImage, "CapturedImage.png");
+            saveImage(realImage, "RealImage.png");
+            throw new RuntimeException(errorMessage);
+        }
+    }
+
+    private static boolean compareImages(BufferedImage capturedImg,
+            BufferedImage realImg) {
+        int capturedPixel;
+        int realPixel;
+        int imgWidth = capturedImg.getWidth(null);
+        int imgHeight = capturedImg.getHeight(null);
+
+        if (imgWidth != IMAGE_WIDTH || imgHeight != IMAGE_HEIGHT) {
+            System.out
+                    .println("Captured and real images are different in size");
+            return false;
+        }
+
+        for (int i = 0; i < imgWidth; i++) {
+            for (int j = 0; j < imgHeight; j++) {
+                capturedPixel = capturedImg.getRGB(i, j);
+                realPixel = realImg.getRGB(i, j);
+                if (capturedPixel != realPixel) {
+                    System.out.println("Captured pixel ("
+                            + Integer.toHexString(capturedPixel) + ") at (" + i
+                            + ", " + j + ") is not equal to real pixel ("
+                            + Integer.toHexString(realPixel) + ")");
+                    return false;
+                }
+            }
+        }
+        return true;
+    }
+
+    private static class ImageCanvas extends Canvas {
+        @Override
+        public void paint(Graphics g) {
+            g.drawImage(realImage, 10, 10, this);
+        }
+    }
+
+    private static void saveImage(BufferedImage image, String fileName) {
+        // Save BufferedImage to PNG file
+        try {
+            File file = new File(fileName);
+            System.out.println("Saving image : " + image + " to \n"
+                    + file.getAbsolutePath());
+            ImageIO.write(image, "PNG", file);
```

Comment:
```suggestion
            ImageIO.write(image, "png", new File(fileName));
```
The body can be shortened.

(Usually, the format is in lower case; that's what I've seen the most in tests which save images.)

Does printing the format of the image and the absolute path help? I see no value.